frontend/src/app/projects/models/model-source/t4c/manage-t4c-model/manage-t4c-model.component.ts

```diff
@@ -0,0 +1,200 @@
+import { notNullValidator } from 'src/app/helpers/validators/validators';
+import {
+  SubmitT4CModel,
+  T4CModel,
+  T4CModelService,
+} from 'src/app/projects/models/model-source/t4c/service/t4c-model.service';
+import { ModelService } from 'src/app/services/model/model.service';
+import { ProjectService } from 'src/app/services/project/project.service';
+import {
+  T4CInstance,
+  T4CInstanceService,
+} from 'src/app/services/settings/t4c-instance.service';
+import {
+  T4CRepoService,
+  T4CRepository,
+} from 'src/app/settings/modelsources/t4c-settings/service/t4c-repos/t4c-repo.service';
+
+@Component({
+  selector: 'app-manage-t4c-model',
+  templateUrl: './manage-t4c-model.component.html',
+  styleUrls: ['./manage-t4c-model.component.css'],
+})
+export class ManageT4CModelComponent implements OnInit, OnDestroy {
+  @Input() asStepper?: boolean;
+  @Output() create = new EventEmitter<boolean>();
+
+  editing = false;
+  loading = false;
+
+  t4cInstances: T4CInstance[] | undefined = undefined;
+  get selectedT4CInstance(): T4CInstance | null | undefined {
+    return this.t4cInstances?.find(
+      (instance: T4CInstance) => instance.id == this.form.value.t4cRepositoryId
+    );
+  }
+
+  t4cRepositories: T4CRepository[] | undefined = undefined;
+  get selectedRepository(): T4CRepository | null | undefined {
+    return this.t4cRepositories?.find(
+      (repository: T4CRepository) =>
+        repository.id == this.form.value.t4cRepositoryId
+    );
+  }
+
+  t4cModel?: T4CModel;
+
+  public form = new FormGroup({
+    t4cInstanceId: new FormControl<null | number>(null, [
+      Validators.required,
+      notNullValidator(),
+    ]),
+    t4cRepositoryId: new FormControl<null | number>(null, [
+      Validators.required,
+      notNullValidator(),
+    ]),
+    name: new FormControl('', [Validators.required]),
+  });
+
+  constructor(
+    public projectService: ProjectService,
+    public modelService: ModelService,
+    public t4cInstanceService: T4CInstanceService,
+    public t4cRepositoryService: T4CRepoService,
+    public t4cModelService: T4CModelService,
+    private router: Router,
+    private route: ActivatedRoute,
+    private toastService: ToastService
+  ) {}
+
+  ngOnInit(): void {
+    this.form.disable({ emitEvent: false });
+    this.loading = true;
+    this.updateRepositoryListOnInstanceChange();
+    this.updateProjectListOnRepositoryChange();
+
+    this.fetchT4CInstances()
+      .pipe(switchMap(() => this.patchFormWithExistingT4CModel()))
+      .subscribe(() => {
+        this.loading = false;
+      });
+  }
+
+  fetchT4CInstances(): Observable<T4CInstance[]> {
+    return this.t4cInstanceService.listInstances().pipe(
+      tap((instances) => {
+        this.t4cInstances = instances;
+        this.form.controls.t4cInstanceId.enable({ emitEvent: false });
+        this.form.controls.t4cInstanceId.patchValue(
+          this.form.controls.t4cInstanceId.value
```

Comment:
This part is not really intuitive, couldn't we use `this.form.controls.t4cInstanceId.updateValueAndValidity()` for the same behavior?

Reply:
I think it's not needed at all. Removed it.